### PR TITLE
Middleware: Rewrite tests to use standard library

### PR DESCRIPTION
**Pull request by aknuds1**
**What this PR does / why we need it**:
Rewrite middleware tests to use standard library instead of goconvey. Part of a series of PRs to remove global state from middleware.

I would recommend diffing with a graphical diff tool, since it will make it much easier to see how things actually change.

Sorry for a large PR! Couldn't be helped unfortunately, since many of these tests are intertwined through common dependencies, and when you change one you tend to have to change all.

### CODE REVIEW

File: pkg/middleware/middleware_test.go [hunk 1]
```diff
@@ -276,283 +294,334 @@ func TestMiddlewareContext(t *testing.T) {
...
+		sc.fakeReq("GET", "/").exec()
 
-			Convey("Should not init context with user info", func() {
-				So(sc.context.IsSignedIn, ShouldBeFalse)
-				So(sc.context.UserId, ShouldEqual, 0)
-				So(sc.context.UserToken, ShouldBeNil)
-			})
+		assert.False(t, sc.context.IsSignedIn)
+		assert.Equal(t, int64(0), sc.context.UserId)
+		assert.Empty(t, sc.context.UserToken)
```
Comment (wbrowne) [inline pkg/middleware/middleware_test.go:349]: `assert.Nil` would be more correct

Comment (aknuds1) [reply pkg/middleware/middleware_test.go:349]: @wbrowne Checking

**Review by wbrowne — APPROVED** [review]
Looking good 🥇

File: pkg/middleware/middleware_test.go (hunk 1 above)
Comment (aknuds1) [reply pkg/middleware/middleware_test.go:349]: @wbrowne Correct! Mistake on my end.

**Pull request merged**